packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -253,14 +349,21 @@ export class SDKClient {
     );
   }
 
-  async getRecord(transactionResponse: TransactionResponse) {
-    return transactionResponse.getRecord(this.clientMain);
-  }
-
+  /**
+   * Submits an Ethereum transaction, potentially using Hedera File Service (HFS) for large call data.
+   * @param {Uint8Array} transactionBuffer - The transaction data buffer.
+   * @param {string} callerName - The name of the caller for logging purposes.
+   * @param {string} requestId - The request ID for logging purposes.
+   * @param {MirrorNodeClient} mirrorNodeClient - The mirror node client for querying transaction records.
+   * @returns {Promise<{ txResponse: TransactionResponse; fileId: FileId | null }>} The transaction response and the file ID if call data is stored in HFS.
+   * @throws {SDKClientError} Throws an SDK client error if the transaction fails or if a file ID is not created for large call data.
+   * @throws {HBAR_RATE_LIMIT_PREEMTIVE_EXCEEDED} Throws if the preemptive HBAR rate limit is exceeded due to large call data.
+   */
   async submitEthereumTransaction(
     transactionBuffer: Uint8Array,
     callerName: string,
     requestId: string,
+    mirrorNodeClient: MirrorNodeClient,
```

Comment:
It seems wrong to couple the `SdkClient` class with the `MirrorNodeClient`. If this is passed because we want to use it for the transaction record retrieval, we can wrap this logic in a new `TransactionService` which delegates to the `SdkClient` and `MirrorNodeClient` as needed and keep the two clients completely separated.

Reply:
Nice I like this! Thanks for the suggestion!

Reply:
resolved with new updates